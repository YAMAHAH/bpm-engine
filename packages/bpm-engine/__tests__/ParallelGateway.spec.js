import fs from 'fs';

import BPMEngine from 'bpm-engine';
import History from './Plugins/History';

describe('ParallelGateway', () => {
  it('Continues on all paths', async (done) => {
    const history = new History();
    const bpm = new BPMEngine({
      plugins: [history],
    });

    const token = await bpm.createProcessInstance({
      workflowDefinition: fs.readFileSync(`${__dirname}/diagrams/ParallelGateway.bpmn`, 'utf-8'),
    });

    await token.execute();
    setTimeout(() => {
      expect(history.store).toMatchSnapshot();
      done();
    }, 100);
  });

  it('Runs services in parallel', async (done) => {
    const history = new History();
    const bpm = new BPMEngine({
      plugins: [history],
    });

    const token = await bpm.createProcessInstance({
      workflowDefinition: fs.readFileSync(`${__dirname}/diagrams/ParallelServices.bpmn`, 'utf-8'),
    });

    await token.execute();
    setTimeout(() => {
      expect(history.store).toMatchSnapshot();
      done();
    }, 100);
  });
});
