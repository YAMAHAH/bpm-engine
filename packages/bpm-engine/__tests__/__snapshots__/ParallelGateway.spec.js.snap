// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParallelGateway Continues on all paths 1`] = `
Array [
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_1fbudxi",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_1fbudxi",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_1fbudxi",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "ParallelGateway_1dy6xai",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "ParallelGateway_1dy6xai",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "ParallelGateway_1dy6xai",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_0ev7re3",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_0ev7re3",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_0ev7re3",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_1bjr67q",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_1bjr67q",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_1bjr67q",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_06106zx",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_06106zx",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_06106zx",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "ParallelGateway_0vuadku",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_06wvy5q",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_06wvy5q",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_06wvy5q",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_10jy1w4",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_10jy1w4",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_10jy1w4",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_1o1qpdi",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_1o1qpdi",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_1o1qpdi",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "ParallelGateway_0vuadku",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "ParallelGateway_0vuadku",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "ParallelGateway_0vuadku",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_0x5p3hw",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_0x5p3hw",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_0x5p3hw",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "EndEvent_13nsh2q",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "EndEvent_13nsh2q",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "EndEvent_13nsh2q",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
]
`;

exports[`ParallelGateway Runs services in parallel 1`] = `
Array [
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_10t4u0l",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_10t4u0l",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_10t4u0l",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_0rnu0cc",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_0rnu0cc",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_0rnu0cc",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_1fyk49e",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_1fyk49e",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_1fyk49e",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "ExclusiveGateway_03c6fwo",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "ExclusiveGateway_03c6fwo",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "ExclusiveGateway_03c6fwo",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_1jkb071",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_1jkb071",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_1jkb071",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_0z13d9i",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_0z13d9i",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_0z13d9i",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_1y4firo",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_1y4firo",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_1y4firo",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "ExclusiveGateway_0jwxcj8",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_0ib5ljr",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_0ib5ljr",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_0ib5ljr",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_1ro3bn5",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_1ro3bn5",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_1ro3bn5",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_1rhm6v9",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_1rhm6v9",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_1rhm6v9",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "ExclusiveGateway_0jwxcj8",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "ExclusiveGateway_0jwxcj8",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "ExclusiveGateway_0jwxcj8",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_1goe8ay",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_1goe8ay",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_1goe8ay",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_0ihlvap",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_0ihlvap",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_0ihlvap",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_0xad4bv",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_0xad4bv",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_0xad4bv",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "EndEvent_15k76o1",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "EndEvent_15k76o1",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "EndEvent_15k76o1",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
]
`;
